tests/test_premium.py

```diff
@@ -0,0 +1,422 @@
+        assert mock_stripe.Subscription.retrieve.called
+
+
+def test_handle_subscription_created(app: Flask) -> None:
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        user.stripe_customer_id = "cus_123"
+        db.session.add(user)
+        db.session.commit()
+
+        subscription = MagicMock()
+        subscription.customer = "cus_123"
+        subscription.id = "sub_123"
+        subscription.status = StripeSubscriptionStatusEnum.INCOMPLETE
+        subscription.cancel_at_period_end = False
+        subscription.current_period_end = datetime.now() + timedelta(days=30)
+        subscription.current_period_start = datetime.now()
+
+        handle_subscription_created(subscription)
+
+        assert user.stripe_subscription_id == "sub_123"
+        assert user.tier_id != BUSINESS_TIER
+
+
+def test_handle_subscription_updated_upgrade(app: Flask) -> None:
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        user.stripe_subscription_id = "sub_123"
+        db.session.add(user)
+        db.session.commit()
+
+        subscription = MagicMock()
+        subscription.id = "sub_123"
+        subscription.status = StripeSubscriptionStatusEnum.ACTIVE
+        subscription.cancel_at_period_end = False
+        subscription.current_period_end = datetime.now() + timedelta(days=30)
+        subscription.current_period_start = datetime.now()
+
+        handle_subscription_updated(subscription)
+
+        assert user.tier_id == BUSINESS_TIER
+
+
+def test_handle_subscription_updated_downgrade(app: Flask) -> None:
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        user.stripe_subscription_id = "sub_123"
+        db.session.add(user)
+        db.session.commit()
+
+        subscription = MagicMock()
+        subscription.id = "sub_123"
+        subscription.status = StripeSubscriptionStatusEnum.CANCELED
+        subscription.cancel_at_period_end = True
+        subscription.current_period_end = datetime.now() + timedelta(days=30)
+        subscription.current_period_start = datetime.now()
+
+        handle_subscription_updated(subscription)
+
+        assert user.tier_id == FREE_TIER
+
+
+def test_handle_subscription_deleted(app: Flask) -> None:
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        user.stripe_subscription_id = "sub_123"
+        db.session.add(user)
+        db.session.commit()
+
+        subscription = MagicMock()
+        subscription.id = "sub_123"
+
+        handle_subscription_deleted(subscription)
+
+        assert user.tier_id == FREE_TIER
+        assert user.stripe_subscription_id is None
+
+
+def test_handle_invoice_created(app: Flask) -> None:
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
```

Comment:
`user` fixture

Reply:
Fixed in 180d5b3514b575c43e6b99320d8471a6f341796f